features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/CustomizedStepper.jsx

```diff
@@ -0,0 +1,340 @@
+        marginLeft: 0,
+        marginRight: theme.spacing(4.2),
+        marginTop: theme.spacing(0.5),
+    },
+    iconTrue: {
+        color: green[500],
+        display: 'block',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        marginRight: theme.spacing(1),
+    },
+    iconFalse: {
+        color: theme.palette.grey[500],
+        display: 'block',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        marginRight: theme.spacing(1),
+    },
+    arrowIcon: {
+        position: 'absolute',
+        top: '16px',
+        right: '-76px',
+        fontSize: '7.9461rem',
+        color: theme.palette.background.default,
+        zIndex: '1',
+    },
+    label: {
+        paddingLeft: '0',
+        paddingRight: '0',
+    },
+    stepper: {
+        background: theme.palette.background.default,
+        marginLeft: theme.spacing(8),
+    },
+    box: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        background: 'white',
+        padding: '2px',
+        width: '405px',
+        height: '82px',
+        borderRight: '0',
+        marginRight: '0',
+    },
+}));
+
+/**
+ * This component renders the API steps
+ *
+ */
+export default function CustomizedSteppers() {
+    const restApi = new Api();
+    const [api, updateAPI] = useAPI();
+    const classes = useStyles();
+    const isEndpointAvailable = api.endpointConfig !== null;
+    const isTierAvailable = api.policies.length !== 0;
+    const isPrototypedAvailable =
+        api.endpointConfig !== null && api.endpointConfig.implementation_status === 'prototyped';
+    const [lifecycleState, setlifecycleState] = useState([]);
+    const [isUpdating, setUpdating] = useState(false);
+
+    useEffect(() => {
+        restApi.getLcState(api.id)
+            .then((result) => {
+                setlifecycleState(result.body.state);
+            });
+    }, []);
+
+    /**
+ * Update the LifeCycle state of the API
+ *
+ */
+    function updateLCStateOfAPI() {
+        setUpdating(true);
+        const promisedUpdate = restApi.updateLcState(api.id, 'Publish');
+        promisedUpdate
+            .then(() => {
+                updateAPI()
+                    .then(() => Alert.info('API Updated Successfully'))
+                    .catch((error) => {
+                        if (error.response) {
+                            Alert.error(error.response.body.description);
+                        } else {
+                            Alert.error('Something went wrong while updating the API');
+                        }
+                        console.error(error);
+                    });
+                Alert.info('Lifecycle state updated successfully');
+            })
+            .catch((errorResponse) => {
+                console.log(errorResponse);
+                Alert.error(JSON.stringify(errorResponse.message));
+            })
+            .finally(() => setUpdating(false));
```

Comment:
Shall we move this final call to `updateAPI()` promise

Reply:
fixed